### PR TITLE
Feature - import API Management policy to a product script

### LINKED ISSUE #15: Provide script to import policy to a product in Azure API Management
Provide script to import policy to a product in Azure API Management

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #15

### CODE REVIEW

File: docs/preview/features/powershell/azure-api-management.md [hunk 1]
```diff
@@ -44,3 +44,19 @@ Creates a new API operation on the API Management service with using a specific
 PS> Create-AzApiManagementApiOperation -ResourceGroup $ResourceGroup -ServiceName $ServiceName -ApiId $ApiId -OperationId $OperationId -Method $Method -UrlTemplate $UrlTemplate -OperationName $OperationName -Description $Description -PolicyFilePath $PolicyFilePath
 # New API operation '$OperationName' on API Management service was added.
 ```	
+
+## Import policy to an product in the API Management service
+
+Imports a policy from a file to an product in Azure API Management.
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-api-management.md:50]:
```suggestion
Imports a policy from a file to a product in Azure API Management.
```

File: docs/preview/features/powershell/azure-api-management.md [hunk 2]
```diff
@@ -44,3 +44,19 @@ Creates a new API operation on the API Management service with using a specific
 PS> Create-AzApiManagementApiOperation -ResourceGroup $ResourceGroup -ServiceName $ServiceName -ApiId $ApiId -OperationId $OperationId -Method $Method -UrlTemplate $UrlTemplate -OperationName $OperationName -Description $Description -PolicyFilePath $PolicyFilePath
 # New API operation '$OperationName' on API Management service was added.
 ```	
+
+## Import policy to an product in the API Management service
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-api-management.md:48]:
```suggestion
## Import policy to a product in the API Management service
```

File: src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1 [hunk 3]
```diff
@@ -48,3 +48,35 @@ function Create-AzApiManagementApiOperation {
 }
 
 Export-ModuleMember -Function Create-AzApiManagementApiOperation
+
+<#
+ .Synopsis
+  Import a policy to an product in Azure API Management.
```
Comment (mbraekman) [inline src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1:54]:
```suggestion
  Import a policy to a product in Azure API Management.
```

File: src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1 [hunk 4]
```diff
@@ -48,3 +48,35 @@ function Create-AzApiManagementApiOperation {
...
+
+ .Description
+  Import a policy to an API scope in a Azure API Management service.
```
Comment (mbraekman) [inline src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1:57]:
```suggestion
  Import a policy to a product in Azure API Management.
```

File: src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1 [hunk 5]
```diff
@@ -48,3 +48,35 @@ function Create-AzApiManagementApiOperation {
...
+
+ .Parameter ResourceGroupName
+  The resource group containing the API Management service.
+
+ .Parameter ServiceName
+  The name of the API Management service located in Azure.
+
+ .Parameter ProductId
+  The ID to identify the product running in API Management.
```
Comment (mbraekman) [inline src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1:66]: It's not really something that is running in APIM, it's more of a identifier/shared configuration into which API's can be added.
```suggestion
  The ID to identify the product in API Management.
```

File: docs/preview/features/powershell/azure-api-management.md [hunk 6]
```diff
@@ -44,3 +44,19 @@ Creates a new API operation on the API Management service with using a specific
...
+
+Imports a policy from a file to an product in Azure API Management.
+
+| Parameter           | Mandatory | Description                                                                                              |
+| ------------------- | --------- | -------------------------------------------------------------------------------------------------------- |
+| `ResourceGroupName` | yes       | The resource group containing the API Management service                                                 |
+| `ServiceName`       | yes       | The name of the API Management service located in Azure                                                  |
+| `ProductId`         | yes       | The ID to identify the product running in API Management                                                     |
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-api-management.md:56]:
```suggestion
| `ProductId`         | yes       | The ID to identify the product in API Management                                                     |
```

**Comment by stijnmoreels** [comment]
> The RequiredModules in the psd1-file should be `Az.ApiManagement` instead of `Az`, no?

Done!

File: src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1 [hunk 7]
```diff
@@ -51,13 +51,45 @@ Export-ModuleMember -Function Create-AzApiManagementApiOperation
...
+<#
+.Synopsis
   Import a policy to an API in Azure API Management.
 
  .Description
   Import a base-policy to an API hosted in Azure API Management.
 
  .Parameter ResourceGroup
-  The resource group containing the API Management service.
+ The resource group containing the API Management service.
```
Comment (tomkerkhove) [inline src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1:92]: Let's be consistent with the docs above
```suggestion
 The resource group containing the Azure API Management service.
```

**Comment by tomkerkhove** [comment]
> Can you change all `API Management` references to include `Azure` as it's a product name please?

There are a couple more places with these btw

**Comment by stijnmoreels** [comment]
> > Can you change all `API Management` references to include `Azure` as it's a product name please?
> 
> There are a couple more places with these btw

Searched and renamed as many as I could find. 🧟‍♂️🤺

File: docs/preview/features/powershell/azure-api-management.md [hunk 8]
```diff
@@ -13,29 +13,29 @@ To have access to the following features, you have to import the module:
...
-Creates a new API operation on the API Management service with using the default base operation policy.
+Creates a new API operation on the Azure API Management service with using the default base operation policy.
 
 ```powershell
 PS> Create-AzApiManagementApiOperation -ResourceGroup $ResourceGroup -ServiceName $ServiceName -ApiId $ApiId -OperationId $OperationId -Method $Method -UrlTemplate $UrlTemplate
-# New API operation '$OperationName' on API Management service was added.
+# New API operation '$OperationName' on Azure API Management service was added.
 ```
 
 Creates a new API operation on the API Management service with using a specific operation policy.
```
Comment (tomkerkhove) [inline docs/preview/features/powershell/azure-api-management.md:41]:
```suggestion
Creates a new API operation on the Azure API Management instance with using a specific operation policy.
```

File: docs/preview/features/powershell/azure-api-management.md [hunk 9]
```diff
@@ -45,32 +45,48 @@ PS> Create-AzApiManagementApiOperation -ResourceGroup $ResourceGroup -ServiceNam
 # New API operation '$OperationName' on API Management service was added.
 ```
```
Comment (tomkerkhove) [inline docs/preview/features/powershell/azure-api-management.md:46]: This looks like a doc bug or not?
```suggestion
```

File: docs/preview/features/powershell/azure-api-management.md [hunk 10]
```diff
@@ -45,32 +45,48 @@ PS> Create-AzApiManagementApiOperation -ResourceGroup $ResourceGroup -ServiceNam
 # New API operation '$OperationName' on API Management service was added.
```
Comment (tomkerkhove) [inline docs/preview/features/powershell/azure-api-management.md:45]: This looks like a doc bug or not?
```suggestion
```

File: docs/preview/features/powershell/azure-api-management.md [hunk 11]
```diff
@@ -45,32 +45,48 @@ PS> Create-AzApiManagementApiOperation -ResourceGroup $ResourceGroup -ServiceNam
...
-| ------------------- | --------- | -------------------------------------------------------------------------------------------------------- |
-| `ResourceGroupName` | yes       | The resource group containing the API Management service                                                 |
-| `ServiceName`       | yes       | The name of the API Management service located in Azure                                                  |
-| `ApiId`             | yes       | The ID to identify the API running in API Management                                                     |
-| `PolicyFilePath`    | yes       | The path to the file containing the to-be-imported policy                                                |
+| Parameter           | Mandatory | Description                                               |
+| ------------------- | --------- | --------------------------------------------------------- |
+| `ResourceGroupName` | yes       | The resource group containing the Azure API Management service  |
+| `ServiceName`       | yes       | The name of the API Management service located in Azure   |
```
Comment (tomkerkhove) [inline docs/preview/features/powershell/azure-api-management.md:71]:
```suggestion
| `ServiceName`       | yes       | The name of the Azure API Management instance located in Azure   |
```

File: src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1 [hunk 12]
```diff
@@ -51,16 +51,48 @@ Export-ModuleMember -Function Create-AzApiManagementApiOperation
...
+  Import a policy to a product in Azure API Management.
+
+ .Description
+  Import a policy to a product in Azure API Management.
+
+ .Parameter ResourceGroupName
+  The resource group containing the Azure API Management service.
+
+ .Parameter ServiceName
+  The name of the API Management service located in Azure.
```
Comment (tomkerkhove) [inline src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1:63]:
```suggestion
  The name of the Azure API Management instance located in Azure.
```

File: src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1 [hunk 13]
```diff
@@ -51,16 +51,48 @@ Export-ModuleMember -Function Create-AzApiManagementApiOperation
 
 <#
  .Synopsis
+  Import a policy to a product in Azure API Management.
+
+ .Description
+  Import a policy to a product in Azure API Management.
+
+ .Parameter ResourceGroupName
+  The resource group containing the Azure API Management service.
```
Comment (tomkerkhove) [inline src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1:60]:
```suggestion
  The resource group containing the Azure API Management instance.
```

File: src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1 [hunk 14]
```diff
@@ -89,13 +121,13 @@ Export-ModuleMember -Function Import-AzApiManagementApiPolicy
   Imports a policy from a file to an API operation in Azure API Management.
 
  .Parameter ResourceGroup
-  The resource group containing the API Management service.
+  The resource group containing the Azure API Management service.
 
  .Parameter ServiceName
   The name of the API Management service located in Azure.
```
Comment (tomkerkhove) [inline src/Arcus.Scripting.ApiManagement/Arcus.Scripting.ApiManagement.psm1:127]:
```suggestion
  The name of the Azure API Management instance located in Azure.
```

File: docs/preview/features/powershell/azure-api-management.md (hunk 9 above)
Comment (stijnmoreels) [reply docs/preview/features/powershell/azure-api-management.md:46]: No, it's the powershell comment that is here presented as the output of the command 😃 .

File: docs/preview/features/powershell/azure-api-management.md (hunk 10 above)
Comment (stijnmoreels) [reply docs/preview/features/powershell/azure-api-management.md:45]: No, it's the powershell comment that is here presented as the output of the command 😃 .

**Pull request merged**